fix: Hides nodes in toolbar when nodes disabled

afbora opened the pull request:
## Description

I fixed the issue via solving passing nodes and headings props to toolbar component. I think these params were missing.

## Changelog
<!--
Add relevant release notes: Features, Enhancements, Fixes, Deprecated.
Reference issues from the `kirby` repo or ideas from `feedback.getkirby.com`.
Always mention whether your PR introduces breaking changes.
-->

### Fixes
- [v5] Writer field without nodes and marks still shows toolbar #7305



### Breaking changes
None


## Docs
<!--
Add any notes that help to document the feature/changes. Doesn't need
to be your best writing, just a few words and/or code snippets.
-->



## Ready?
<!--
If you can help to check off the following tasks, that'd be great.
If not, don't worry - we will take care of it.
-->

- [x] In-code documentation (wherever needed)
- [ ] Unit tests for fixed bug/feature
- [x] Tests and CI checks all pass

### For review team
<!--
We will take care of the following before merging the PR.
-->

- [ ] Add lab and/or sandbox examples (wherever helpful)
- [ ] Add changes & docs to release notes draft in Notion

afbora committed:
fix(#7305): Hides nodes in toolbar when nodes disabled

distantnative commented on panel/src/components/Forms/Writer/Toolbar.vue:
```diff
@@ -297,6 +304,25 @@ export default {
 				click: () => this.command(entry.command ?? type)
 			};
 		},
+		/**
+		 * Helper method to expand node types
+		 * @returns {Array}
+		 */
+		expandNodeTypes(nodes) {
+			const expanded = [];
+
+			nodes.forEach((node) => {
+				if (node === 'heading') {
+					this.headings.forEach((level) => {
+						expanded.push(`h${level}`);
+					});
+				} else {
+					expanded.push(node);
+				}
+			});
+
+			return expanded;
+		},
```
`Todo:` I do not think we need to expand the headings here. The `Heading` node itself is registering multiple buttons depending on the heading level. Did this not work for you or why did you add this new helper method?

afbora replied:
It is set as `heading` index in the `nodes` array. But in the `available` array, it is passed as "h1-h6" indexes instead of `heading`. So I had to convert `heading` to `h*` indexdes.

distantnative replied:
But something is off here, the headings node entries show up in different order before/after the PR. I am not sure I can fully follow yet why this change is needed.

distantnative reviewed (commented):
I think we need to move this to 5.0.1 or so - there are some changes needed. See one comment I made, but also:

`Todo:` With this PR the paragraph node is gone from the dropdown when not having any `nodes` option set. I don't think this is intended or is it?

distantnative commented on panel/src/components/Forms/Input/WriterInput.vue:
```diff
@@ -141,9 +141,14 @@ export default {
+					(Array.isArray(this.nodes) || this.nodes === false)
+						? this.nodes
+						: undefined,
+				headings: this.headings,
```
I do not think we need to pass `headings` to the toolbar. We pass it to the `Heading` node which then takes care itself which heading levels to show as node buttons.

afbora committed:
fix: Add 'paragraph' to default nodes in WriterInput

Updated the default value of the 'nodes' prop in WriterInput.vue to include 'paragraph' alongside 'heading', 'bulletList', and 'orderedList'. This ensures paragraph nodes are available by default.
